src/Features/Core/Portable/MetadataAsSource/DecompilationMetadataAsSourceFileProvider.cs

```diff
@@ -171,34 +170,26 @@ internal sealed class DecompilationMetadataAsSourceFileProvider(IImplementationA
             var timeout = TimeSpan.FromSeconds(5);
             var firstAttempt = true;
             var skipWritingFile = false;
-            while (!Directory.Exists(directoryToCreate))
+
+            while (!IOUtilities.PerformIO(() => Directory.Exists(directoryToCreate)))
             {
-                if (stopwatch.Elapsed > TimeSpan.FromSeconds(5))
+                if (stopwatch.Elapsed > timeout)
                 {
                     // If we still can't create the folder after 5 seconds, assume we will not be able to create it.
                     skipWritingFile = true;
                     break;
                 }
 
-                try
+                if (firstAttempt)
                 {
-                    if (firstAttempt)
-                    {
-                        firstAttempt = false;
-                    }
-                    else
-                    {
-                        await Task.Delay(DelayTimeSpan.Short, cancellationToken).ConfigureAwait(false);
-                    }
-
-                    Directory.CreateDirectory(directoryToCreate);
+                    firstAttempt = false;
                 }
-                catch (DirectoryNotFoundException)
-                {
-                }
-                catch (UnauthorizedAccessException)
+                else
                 {
+                    await Task.Delay(DelayTimeSpan.Short, cancellationToken).ConfigureAwait(false);
                 }
+
+                IOUtilities.PerformIO(() => Directory.CreateDirectory(directoryToCreate));
```

Comment:
>CreateDirectory [](http://example.com/codeflow?start=54&length=15)

It doesn't look like IOUtilities.IsNormalIOException includes DirectoryNotFoundException

Reply:
lies.

Reply:
Whoops, cyrus pointed out to me it does.